driver/optcompile.ml

```diff
@@ -38,50 +38,54 @@ let flambda i backend Typedtree.{structure; coercion; _} =
     Clflags.unbox_free_vars_of_closures := false;
     Clflags.unbox_specialised_args := false
   end;
-  (structure, coercion)
-  |> Profile.(record transl)
-      (Translmod.transl_implementation_flambda i.module_name)
-  |> Profile.(record generate)
-    (fun {Lambda.module_ident; main_module_block_size;
-          required_globals; code } ->
-    ((module_ident, main_module_block_size), code)
-    |>> print_if i.ppf_dump Clflags.dump_rawlambda Printlambda.lambda
-    |>> Simplif.simplify_lambda
-    |>> print_if i.ppf_dump Clflags.dump_lambda Printlambda.lambda
-    |> (fun ((module_ident, main_module_block_size), code) ->
-      let program : Lambda.program =
-        { Lambda.
-          module_ident;
-          main_module_block_size;
-          required_globals;
-          code;
-        }
-      in
-      Asmgen.compile_implementation
-        ~backend
-        ~prefixname:i.output_prefix
-        ~middle_end:Flambda_middle_end.lambda_to_clambda
-        ~ppf_dump:i.ppf_dump
-        program);
-    Compilenv.save_unit_info (cmx i))
+  if Clflags.(should_stop_after Compiler_pass.Lambda) then () else begin
```

Comment:
I suspect that the code to support this option in the native compiler (here) may be incorrect. Could you tweak the test you added (thanks!) to try both ocamlc and ocamlopt? (If the output differs it is possible to use different reference files, `foo.byte.reference` and `foo.opt.reference .)